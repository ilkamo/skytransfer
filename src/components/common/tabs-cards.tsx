import {
    Tabs,
} from 'antd';

const { TabPane } = Tabs;

type TabCardValue = {
    name: string
    content: React.ReactNode
}

type TabCardProps = {
    values: TabCardValue[]
}

const TabsCards = ({ values }: TabCardProps) => {
    const components = values.map((value: TabCardValue, index: number) => <TabPane tab={value.name} key={index}>{value.content}</TabPane>);

    return <Tabs type="card">{components}</Tabs>
}

export default TabsCards;