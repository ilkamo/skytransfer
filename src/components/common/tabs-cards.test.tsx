import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { TabsCards } from './tabs-cards';

describe('TabsCards', () => {
  test('renders correctly when there are no values', () => {
    const { container } = render(<TabsCards values={[]} />);

    expect(container).toMatchSnapshot();
  });

  test('renders correctly with values', () => {
    const values = [
      {
        name: 'Some name 1',
        content: <div>Some content 1</div>,
      },
      {
        name: 'Some name 2',
        content: <div>Some content 2</div>,
      },
    ];

    const { container } = render(<TabsCards tabType="line" values={values} />);

    expect(container).toMatchSnapshot();
  });
});
