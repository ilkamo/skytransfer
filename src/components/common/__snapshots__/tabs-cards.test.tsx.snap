// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsCards renders correctly when there are no values 1`] = `
<div>
  <div
    class="ant-tabs ant-tabs-top ant-tabs-card"
  >
    <div
      class="ant-tabs-nav"
      role="tablist"
    >
      <div
        class="ant-tabs-nav-wrap"
      >
        <div
          class="ant-tabs-nav-list"
          style="transform: translate(0px, 0px);"
        >
          <div
            class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
          />
        </div>
      </div>
      <div
        class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
      >
        <button
          aria-controls="rc-tabs-test-more-popup"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-hidden="true"
          class="ant-tabs-nav-more"
          id="rc-tabs-test-more"
          style="visibility: hidden; order: 1;"
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="ant-tabs-content-holder"
    >
      <div
        class="ant-tabs-content ant-tabs-content-top"
      />
    </div>
  </div>
</div>
`;

exports[`TabsCards renders correctly with values 1`] = `
<div>
  <div
    class="ant-tabs ant-tabs-top ant-tabs-card"
  >
    <div
      class="ant-tabs-nav"
      role="tablist"
    >
      <div
        class="ant-tabs-nav-wrap"
      >
        <div
          class="ant-tabs-nav-list"
          style="transform: translate(0px, 0px);"
        >
          <div
            class="ant-tabs-tab ant-tabs-tab-active"
          >
            <div
              aria-controls="rc-tabs-test-panel-0"
              aria-selected="true"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-0"
              role="tab"
              tabindex="0"
            >
              Some name 1
            </div>
          </div>
          <div
            class="ant-tabs-tab"
          >
            <div
              aria-controls="rc-tabs-test-panel-1"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-1"
              role="tab"
              tabindex="0"
            >
              Some name 2
            </div>
          </div>
          <div
            class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
          />
        </div>
      </div>
      <div
        class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
      >
        <button
          aria-controls="rc-tabs-test-more-popup"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-hidden="true"
          class="ant-tabs-nav-more"
          id="rc-tabs-test-more"
          style="visibility: hidden; order: 1;"
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="ant-tabs-content-holder"
    >
      <div
        class="ant-tabs-content ant-tabs-content-top"
      >
        <div
          aria-hidden="false"
          aria-labelledby="rc-tabs-test-tab-0"
          class="ant-tabs-tabpane ant-tabs-tabpane-active"
          id="rc-tabs-test-panel-0"
          role="tabpanel"
          tabindex="0"
        >
          <div>
            Some content 1
          </div>
        </div>
        <div
          aria-hidden="true"
          aria-labelledby="rc-tabs-test-tab-1"
          class="ant-tabs-tabpane"
          id="rc-tabs-test-panel-1"
          role="tabpanel"
          style="display: none;"
          tabindex="-1"
        />
      </div>
    </div>
  </div>
</div>
`;
